src/renderer/components/+preferences/application.tsx

```diff
@@ -135,16 +128,9 @@ export const Application = observer(() => {
 
       <section id="other">
         <SubTitle title="Start-up"/>
-        <FormSwitch
-          control={
-            <Switcher
-              checked={userStore.openAtLogin}
-              onChange={v => userStore.openAtLogin = v.target.checked}
-              name="startup"
-            />
-          }
-          label="Automatically start Lens on login"
-        />
+        <Switch checked={userStore.openAtLogin} onChange={v => userStore.openAtLogin = v.target.checked}>
```

Comment:
```suggestion
        <Switch checked={userStore.openAtLogin} onChange={(enabled: boolean, event: MouseEvent) => userStore.openAtLogin = v.target.checked}>
```

_DRY_: i think that component should receive actual boolean value as first argument.

Reply:
Why it needs to be added? Also, this change would collide with native `onChange` types.
```
interface Props extends React.HTMLProps<HTMLInputElement> {
}
```

Reply:
Exclude native `onChange` from interface and define yours (see: `checkbox.ts` for example)
Or you can even skip extending from `React.HTMLProps` and keep only necessary props.

Reply:
_Reason:_ to avoid repeating boilerplate / taking value from event since in most cases we just need current state of the component (not mouse event)

Reply:
I don't agree we should juggle with built-in properties. 
1. The Switch component is basically `<input type="checkbox"/>` and should have all same properties.
2. We shouldn't rely on value from `onChange` callback at all because we're using controlled component model. Since we're passing `checked` prop explicitly, we must preserve single source of truth and get value from stores, not from the `Switch`.

I've made changes replacing this one
```
<Switch checked={userStore.terminalCopyOnSelect} onChange={v => userStore.terminalCopyOnSelect = v.target.checked}>
  Copy on select and paste on right-click
</Switch>
```

with this one

```
<Switch checked={userStore.terminalCopyOnSelect} onChange={() => userStore.terminalCopyOnSelect = !userStore.terminalCopyOnSelect}>
  Copy on select and paste on right-click
</Switch>
```

Reply:
> Since we're passing checked prop explicitly, we must preserve single source of truth and get value from stores, not from the Switch.

Didn't get what you mean.. Why it's single source of truth in stores?? when you change something it comes from user event right? So you getting new source of truth always first from that user event. Wdyt?

> I don't agree we should juggle with built-in properties.

Just consider `<Switch/>` as custom component, not just  plain html`<NewCheckbox/>`, it's not juggling, it's a matter of convenient API for end-users (of this component, remember this is gonna be exported to extensions-api). 
And it's  actually simple to implement: pass in callback from `props.checked`, what's the issue with that?

Reply:
Done with
```
interface Props extends Omit<HTMLProps<HTMLInputElement>, "onChange"> {
  onChange?: (checked: boolean, event: ChangeEvent<HTMLInputElement>) => void;
}
```